### PR TITLE
[test] add code spell check and correct wrong spelling

### PR DESCRIPTION
**Pull request by zhanglongxia**

### CODE REVIEW

File: src/cli/README_HISTORY.md [hunk 1]
```diff
@@ -459,11 +459,11 @@ Print the IPv6 message RX history as a table:
 |         00:00:30.000 | src: [fe80:0:0:0:d03d:d3e7:cc5e:7cd7]:19788                                 |
 |                      | dst: [fe80:0:0:0:a4a5:bbac:a8e:bd07]:19788                                  |
 +----------------------+------------------+-------+--------+-----+------+------+----+--------+-------+
-|                      | UDP              |    93 | 0x52df |  no |  net |  -20 | RX | unknwn |  15.4 |
+|                      | UDP              |    93 | 0x52df |  no |  net |  -20 | RX | unknown|  15.4 |
```
Comment (abtink) [inline src/cli/README_HISTORY.md:462]: This is not a typo (we use shorter string to keep table column width smaller).

I think we should not check the `cli` output for typo.

File: src/cli/README_HISTORY.md [hunk 2]
```diff
@@ -459,11 +459,11 @@ Print the IPv6 message RX history as a table:
...
 |         00:00:30.480 | src: [fe80:0:0:0:d03d:d3e7:cc5e:7cd7]:19788                                 |
 |                      | dst: [fe80:0:0:0:a4a5:bbac:a8e:bd07]:19788                                  |
 +----------------------+------------------+-------+--------+-----+------+------+----+--------+-------+
-|                      | UDP              |    50 | 0x5ccf |  no |  net |  -20 | RX | unknwn |  15.4 |
+|                      | UDP              |    50 | 0x5ccf |  no |  net |  -20 | RX | unknown|  15.4 |
```
Comment (abtink) [inline src/cli/README_HISTORY.md:466]: Same here

File: src/cli/cli_history.cpp [hunk 3]
```diff
@@ -620,7 +620,7 @@ void History::OutputRxTxEntryTableFormat(const otHistoryTrackerMessageInfo &aInf
     }
     else if (aInfo.mNeighborRloc16 == kShortAddrInvalid)
     {
-        OutputFormat("| unknwn ");
+        OutputFormat("| unknown ");
```
Comment (abtink) [inline src/cli/cli_history.cpp:623]: This is not a typo  and intentional so that it fist in the table column width.

**Review by abtink — COMMENTED** [review]
Thanks. This would be very useful indeed. 👍 
Couple of suggestions below:

File: src/cli/README_HISTORY.md (hunk 1 above)
Comment (zhanglongxia) [reply src/cli/README_HISTORY.md]: I have added the word `unknwn` to the config file of the code spell. Then the word `unknwn` will not be considered a wrong word anymore.

**Review by abtink — APPROVED** [review]
This will be very helpful. Thanks. 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**